### PR TITLE
Use `readData` to read data from a pipe instead of `availableData`

### PR DESCRIPTION
**Pull request by ahoppen**
Analogous fix to https://github.com/swiftlang/swift-package-manager/pull/8047

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/TSCBasic/Process/Process.swift [hunk 1]
```diff
@@ -575,7 +575,7 @@ public final class Process {
 
             group.enter()
             stdoutPipe.fileHandleForReading.readabilityHandler = { (fh : FileHandle) -> Void in
-                let data = fh.availableData
+                let data = fh.readData(ofLength: Int.max)
```
Comment (MaxDesiatov) [inline Sources/TSCBasic/Process/Process.swift:578]: Same as on the SwiftPM, this should rely on a non-deprecated method.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**